Source/WebCore/css/CSSFontFaceSet.cpp

```diff
@@ -169,30 +168,28 @@ String CSSFontFaceSet::familyNameFromPrimitive(const CSSPrimitiveValue& value)
 
 void CSSFontFaceSet::addToFacesLookupTable(CSSFontFace& face)
 {
-    auto families = face.families();
-    if (!families) {
+    auto family = face.familyCSSValue();
+    if (!family) {
         // If the font has failed, there's no point in actually adding it to m_facesLookupTable,
         // because no font requests can actually use it for anything. So, let's just ... not add it.
         return;
     }
 
-    for (auto& item : *families) {
-        auto familyName = AtomString { CSSFontFaceSet::familyNameFromPrimitive(downcast<CSSPrimitiveValue>(item)) };
-        if (familyName.isNull())
-            continue;
-
-        auto addResult = m_facesLookupTable.add(familyName, Vector<Ref<CSSFontFace>>());
-        auto& familyFontFaces = addResult.iterator->value;
-        if (addResult.isNewEntry) {
-            // m_locallyInstalledFontFaces grows without bound, eventually encorporating every font installed on the system.
-            // This is by design.
-            if (m_owningFontSelector)
-                ensureLocalFontFacesForFamilyRegistered(familyName);
-            familyFontFaces = { };
-        }
+    auto familyName = AtomString { CSSFontFaceSet::familyNameFromPrimitive(downcast<CSSPrimitiveValue>(*family)) };
```

Comment:
What guarantees that `familyCSSValue` returns a `CSSPrimitiveValue`?

Reply:
Like the old code, nothing but the guarantees of the parser make this true. (I am sure people are sick of me saying it, but switching to the strong representations will fix this).